Application/Transfer/Import/FSE/Service.php

```diff
@@ -0,0 +1,1116 @@
+                                        . ' Der Ansprechpartner wurde nicht angelegt, da schon eine Person mit gleichen Namen existiert.';
+
+                                    $tblPerson = $tblPersonExits;
+                                } else {
+                                    $salutation = trim($Document->getValue($Document->getCell($Location['Anrede'], $RunY)));
+                                    if ($salutation == 'Herr' || $salutation == 'Herrn') {
+                                        $tblSalutation = Person::useService()->getSalutationById(1);
+                                    } elseif ($salutation == 'Frau') {
+                                        $tblSalutation = Person::useService()->getSalutationById(2);
+                                    } else {
+                                        $tblSalutation = false;
+                                    }
+
+                                    $tblPerson = Person::useService()->insertPerson(
+                                        $tblSalutation ? $tblSalutation : null,
+                                        '',
+                                        $firstName,
+                                        '',
+                                        $lastName,
+                                        $groupArray
+                                    );
+                                }
+
+                                if ($tblPerson) {
+                                    Relationship::useService()->addCompanyRelationshipToPerson(
+                                        $tblCompany,
+                                        $tblPerson,
+                                        $tblTypeCommon
+                                    );
+                                }
+                            }
+                        }
+                    }
+
+                    return
+                        new Success('Es wurden ' . $countNewCompany . ' Firmen erfolgreich angelegt.')
+                        . new Success('Es wurden ' . $countEditCompany . ' Firmen erfolgreich umbenannt.')
+                        . new Layout(new LayoutGroup(new LayoutRow(new LayoutColumn(
+                            new Panel(
+                                'Fehler',
+                                $error,
+                                Panel::PANEL_TYPE_DANGER
+                            )
+                        ))));
+
+                } else {
+                    return new Warning(json_encode($Location)) . new Danger(
+                            "File konnte nicht importiert werden, da nicht alle erforderlichen Spalten gefunden wurden");
+                }
+            }
+        }
+
+        return new Danger('File nicht gefunden');
+    }
+
+    /**
+     * @param IFormInterface|null $Form
+     * @param UploadedFile|null $File
+     *
+     * @return IFormInterface|Danger|string
+     */
+    public function createStudentsFromFile(
```

Comment:
Codacy found an issue: [The method createStudentsFromFile() has an NPath complexity of 15583373968. The configured NPath complexity threshold is 200.](https://app.codacy.com/manual/gerdchristian-kunze/SPHERE-Framework/pullRequest?prid=5757982)